### ISSUE TITLE
CRAN task view proposal: ActuarialScience

### ISSUE THREAD

**Issue by dutangc** [cran-task-views/ctv#54]
Dear CTV editors,
With Vincent Goulet, we propose a new task view dedicated to actuarial science. The proposal markdown file is available at [ActuarialScience](https://github.com/dutangc/CTVproposal/blob/main/ActuarialScience.md)

Currently, the Finance task view list some packages linked to actuarial science but there is no comprehensive review of this topic.

The core packages are [actuar](https://cran.r-project.org/package=actuar), [CASdatasets](https://cran.r-project.org/package=CASdatasets), [ChainLadder](https://cran.r-project.org/package=ChainLadder), [lifecontingencies](https://cran.r-project.org/package=lifecontingencies), [mbbefd](https://cran.r-project.org/package=mbbefd), [ReIns](https://cran.r-project.org/package=ReIns).

Any comment is welcome.

**Comment by zeileis** [comment]
Christophe @dutangc & Vincent, thanks for putting this together!

I just had a very brief look and didn't see an obivous discussion of inclusion/exclusion criteria. Could you please expand on this, especially with regard to the separation vis-a-vis the Finance task view. Is this what you want to use the list of references at the beginning for?

Dirk @eddelbuettel, it would be great, if you could have a closer look at this wearing your Finance task view maintainer hat. What do you think about the separation between the two topics?

Technical note: You "can" set explicit anchor points for each section but you do not need to. By default pandoc will add the section titles in lower case with spaces replaced by hyphens, e.g.,  <https://CRAN.R-project.org/web/views/Econometrics.html#instrumental-variables>.

More when I had the chance to take a closer look.

**Comment by eddelbuettel** [comment]
It is pretty long so I could do with a second read too but main 'beef' is editorial:  As @zeileis mentioned you do not need anchors, and having them and the extra brackets makes reading in 'preview' mode more cumbersome than it should be.  In raw code mode it is also a bit strange that you seemingly told your editor to wrap lines at rather unpleasantly low column count (NB: for markdown some people go the other way and never wrap a line to allow per-sentence diffs, that is tricky too. All preferences I suppose).

Overall I think this could make sense and would allow Finance to shrink a litttle.  I am a litte unsure about all the references at the start (it is _CRAN_ task view after all) but if you feel strongly about this maybe keep them.  Otherwise seeing all the references I thought the section on the mortatility data could do a word or two on each line to show the author/agency collection them (ie maybe list mortality.org, U Montreal, ... and state AusHD and FrData) or maybe the short monikers are clear for experts in the field?

**Comment by dutangc** [comment]
Thanks for your feedback.

@zeileis , we put books in the view because some readers of the view do not know them and it is important to look at it before asking questions to maintainers. I did same for https://cran.r-project.org/view=ExtremeValue and https://cran.r-project.org/view=Distributions. However we can put it at the end of the view if needed?

@zeileis , both practitioners and academics do not have the same background and the same topics of interest between Finance and Actuarial science, yet some are common. That's why we consider the view is complementary to the finance view. we can add a link to it at the beginning of the view.

@zeileis , @eddelbuettel , we note that most anchors can be removed from the md file. 

@eddelbuettel , yes we can put some words on mortality data for people who are not aware of the good job done by HMD and other national statistics offices.

@eddelbuettel , with @vigou3, we have the habit to wrap lines. we can write differently if needed.

**Comment by eddelbuettel** [comment]
i) yes move to books to end, consider shortening the section and putting this into a complementary blog post etc, 
ii) a link to the Finance TV strikes me as appropriate 
iii) thanks for nixing anchors 
iv) thanks for edits
v) what horror :grin: esp as we all know that @vigou3 knows how to handle a performant editor -- but "your file, your rules" (subject to basic conformance with CTV best practices)

**Comment by zeileis** [comment]
Just some additional clarifications:

- I fully agree that some references are useful in task view. In the extreme value task view you have one moderately long list at the end. In the distributions task view you have a very short list at the beginning embedded into the text. Both are fine. But in the proposed task view you have several lists with subsections at the beginning without proper embedding into the rest of the text. Putting them at the end would work. But it might also be useful to list two or three references at the beginning of the relevant sections (with context explained in the text).
- As for the inclusion and exclusion criteria: I also agree that finance and actuarial science are sufficiently different to have different task views. But given that there is overlap, where do you draw the line? What is still finance, what is actuarial science? In which (few) cases is a duplicated listing appropriate? Similarly, it could be clarified where you separate between actuarial science and extreme value theory. The latter is a bit easier, though, because you maintain both of the task views.

Furthermore, I noticed:

- An introduction is missing. This should set the scene, discuss inclusion/exclusion criteria (see above), etc. You can also include the acknowledgments in this intro.
- You use hard links to other task views. Please avoid that and use the dedicated `r view("...")` tags instead.
- Add a `### Links` section at the end of the .md. You do mention several relevant links in the text (e.g., mortality.org) and listing the important ones in a dedicated section at the end would be useful, I think.
- You deal with archived packages in a non-standard way. In the new Markdown-based workflow, archived packages are automatically flagged as such in the HTML version. However, we recommend that actions should be taken for these packages and they shouldn't be kept in the task view for more than 100 days. For more details, see the "CRAN package archivals" section in the maintenance documentation: https://github.com/cran-task-views/ctv/blob/main/Maintenance.md#CRAN-package-archivals
- If you want to hyperlink DOIs, you can easily do so with the `r doi("...")` tag.
- You use `> ` quotes for the text that refers to other task views. I don't think this is really necessary. I would put this in regular text.
- I agree with Dirk that the "Mortality databases" section is hard to parse if you miss the package description of `MortalityLaws` in the previous section. I would recommend putting the links into the dedicated `### Links` section at the end of the task view and mention this in the description of the `MortalityLaws` package.

**Comment by dutangc** [comment]
Dear all, 

With Vincent, we make a new proposal for a view on actuarial science after taking into account Dirk and Achim's comments. 

Please take a look at [ActuarialView](https://github.com/dutangc/CTVproposal/blob/main/ActuarialScience.md).

Regards, Christophe

**Comment by zeileis** [comment]
Thanks for all the hard work, Christophe @dutangc and Vincent @vigou3 ! The revised version already looks pretty good to me. I just have a couple of smaller comments, mostly editorial, see below. It would be great if you could also incorporate these.

Thanks & best wishes, Achim

- The `topic` field in the header should be in title case: _Actuarial Science_.
- In the paragraph starting with _"The maintainers gratefully acknowledge..."_ it would be good to also add some encouragement to contribute to the task view with pointers on how this can be done. See some of the other recent task views for examples on how this can be done.
- For the table of contents it's probably easiest to consistently use sentence case. Thus, you need to switch  
  _Life Insurance_ &rightarrow; _Life insurance_  
  _Non Life Insurance_ &rightarrow; _Non-life insurance_  
  Note that the latter should also be hyphenated consistently.
- Some elements of the table of contents are out of sync: _Related links_ should be _Bibliography_ and _Misc_ should be _Miscellaneous_.
- The bibliography is still quite long, maybe it would make sense to cut some of the items that are not directly relevant to R.
- The `view(...)` tag only adds the hyperlinked name of the task view but does not add the words "task view". Your formulations seem to assume otherwise, please adapt correspondingly.
- At the end of the introductory sentence in Life insurance > Loss modeling about the "Distributions" task view, you might add a sentence like: _Here we discuss only those packages that implement distributions particularly designed for actuarial science._
- Please check capitalization also throughout the task view, e.g., _incurred_ or _lognormal_ could probably be lower case? Similarly _conditional tail expectation_ or _actuarial measures_ etc.

**Comment by dutangc** [comment]
Thanks for your feedback. I update the [view](https://github.com/dutangc/CTVproposal/blob/main/ActuarialScience.md) and also shorten the bibliography. You can follow my [todolist](https://github.com/dutangc/CTVproposal/blob/main/TODOlist.md).

The two subsections regarding general references have been limited to recent (comprehensive) books with core applications in actuarial science.

Regards, Christophe

**Comment by zeileis** [comment]
Thanks, Christophe, looks good to me. There is just one "Life Insurance" section title left that should become "Life insurance". I think the CRAN task view is ready for publication now.

@eddelbuettel @tuxette @rsbivand what do you think? Thumbs up is sufficient for endorsement. Further comments are also welcome!

**Comment by dutangc** [comment]
"life insurance" Done!

**Comment by zeileis** [comment]
Thanks for the follow-up, everyone, and apologies for the long delay. I think the task view is ready for release on CRAN.

Christophe @dutangc could you please transfer ownership of the repository to me personally ("zeileis") and I will then transfer ownership to the "cran-task-views" organization and make you co-admin. And then we can work through resolving the last fine details before we release on CRAN.

For transfer of the ownership, the following should work for you:

Settings > Collaborators > Public Repository / Mange > Danger Zone / Transfer ownership

**Comment by zeileis** [comment]
Thanks, Christophe! I've made a few small touch-ups and unifications and everything is ready for release, see: https://github.com/cran-task-views/ActuarialScience

If you don't have anything to add, I'll release everything on CRAN.

**Comment by dutangc** [comment]
Ok for me, you can proceed.

**Comment by zeileis** [comment]
Great, it's online now. I'll announce it on social media tomorrow!
https://CRAN.R-project.org/view=ActuarialScience
